wallet-new/src/Cardano/Wallet/API/V1/Errors.hs

```diff
@@ -1,19 +1,104 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
 
 module Cardano.Wallet.API.V1.Errors where
 
 import           Universum
 
-import           Cardano.Wallet.API.V1.Types (WalletError (..))
-import           Data.Aeson (encode)
+import           Data.Aeson
+import           Data.Aeson.TH (deriveJSON)
 import qualified Network.HTTP.Types.Header as HTTP
 import           Servant
+import           Test.QuickCheck (Arbitrary (..), oneof)
 
+import           Cardano.Wallet.API.V1.TH (conNamesList, deriveWalletErrorJSON)
+
+--
+-- Error handling
+--
+
+-- | Type representing any error which might be thrown by wallet.
+--
+-- Errors are represented in JSON in the following way:
+-- ```
+-- {
+--     "message" : <constr_name>,
+--     "diagnostic" : <data>
```

Comment:
It might be nice to cite we didn't invent this but this is actually known as [JSend](https://labs.omniti.com/labs/jsend), to even futher back-up our choice.